app/models/authority.rb

```diff
@@ -340,32 +340,17 @@ def popular_tags_used_on_works_with_count
                                            taggable_id: mm }).group('tags.id').order('count_all DESC').count
   end
 
-  # class variable
-  # rubocop:disable Style/ClassVars
-  @@popular_authors = nil
-
-  def self.recalc_popular
-    evs = Ahoy::Event.where(name: 'view').where("JSON_EXTRACT(properties, '$.type') = 'Authority'").where(
-      'time > ?', 1.month.ago
-    )
-    pop = {}
-    evs.each do |x|
-      au_id = x.properties['id']
-      pop[au_id] = 0 unless pop[au_id].present?
-      pop[au_id] += 1
-    end
-    sorted_pop_keys = pop.keys.sort_by { |k| pop[k] }.reverse
-    Authority.find(sorted_pop_keys[0..9])
-
-    @@popular_authors = has_toc.order(impressions_count: :desc).limit(10).all.to_a
```

Comment:
And here we simply ignored result of previous loop and fetched authors based on impressions_count (i.e. using alltime stats)